walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,139 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.HashSet;
+import java.util.Set;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    Collection<String> optionalAttendees = new ArrayList<String>();
+    // this try-catch avoids a potential NullPointerException
+    try {
+      optionalAttendees = request.getOptionalAttendees();
+    } catch(NullPointerException e) {
+    }
```

Comment:
Nit: This logic looks good, but the `query` function is getting a little long.  I think it might make sense to put these lines 27-32 in a separate function, something like `getOptionalAttendees(request)`, which hides some of the details that `request.getOptionalAttendees();` might throw an exception.

Reply:
Addressed in "created getOptionalAttendees function"